core/src/main/kotlin/com/justai/jaicf/helpers/context/BotContextProperty.kt

```diff
@@ -0,0 +1,139 @@
+package com.justai.jaicf.helpers.context
+
+import com.justai.jaicf.activator.selection.ActivationSelector.Companion.default
+import com.justai.jaicf.context.BotContext
+import kotlin.properties.ReadOnlyProperty
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+/**
+ * An alias for a property delegate of type [V] backed by [BotContext]
+ */
+typealias BotContextProperty<V> = MapBackedProperty<BotContext, V>
+
+/**
+ * Creates a property delegate of type [V] backed by [BotContext.client].
+ *
+ * @param key the key of the entry where to store the property value, if `null` property name is used
+ * @param saveDefault whether to save generated [default] value in the [BotContext.client], `false` by default
+ * @param removeOnNull whether to remove entry from [BotContext.client] on null set, `false` by default
+ * @param default provider of a default value for the entry, [NoSuchElementException] will be thrown by default
+ */
+fun <V> clientProperty(
+    key: String? = null,
+    saveDefault: Boolean = false,
+    removeOnNull: Boolean = false,
+    default: () -> V = { throw NoSuchElementException("No value found for the key specified") }
+) : BotContextProperty<V> = MapBackedProperty(BotContext::client, key, saveDefault, removeOnNull, default)
+
+/**
+ * Creates a property delegate of type [V] backed by [BotContext.session].
+ *
+ * @param key the key of the entry where to store the property value, if `null` property name is used
+ * @param saveDefault whether to save generated [default] value in the [BotContext.session], `false` by default
+ * @param removeOnNull whether to remove entry from [BotContext.session] on null set, `false` by default
+ * @param default provider of a default value for the entry, [NoSuchElementException] will be thrown by default
+ */
+fun <V> sessionProperty(
+    key: String? = null,
+    saveDefault: Boolean = false,
+    removeOnNull: Boolean = false,
+    default: () -> V = { throw NoSuchElementException("No value found for the key specified") }
+) : BotContextProperty<V> = MapBackedProperty(BotContext::session, key, saveDefault, removeOnNull, default)
+
+/**
+ * Creates a property delegate of type [V] backed by [BotContext.temp].
+ *
+ * @param key the key of the entry where to store the property value, if `null` property name is used
+ * @param saveDefault whether to save generated [default] value in the [BotContext.temp], `false` by default
+ * @param removeOnNull whether to remove entry from [BotContext.temp] on null set, `false` by default
+ * @param default provider of a default value for the entry, [NoSuchElementException] will be thrown by default
+ */
+fun <V> tempProperty(
+    key: String? = null,
+    saveDefault: Boolean = false,
```

Comment:
Temp property cannot be saved or removed

Reply:
Can you, please, clarify what do you mean exactly?
I think it can be both saved and removed during single request execution.
It cannot be "persisted", but I don't use this word.